cmd/history.go

```diff
@@ -0,0 +1,115 @@
+// Copyright 2016-2017, Pulumi Corporation.  All rights reserved.
+package cmd
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/dustin/go-humanize"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+
+	"github.com/pulumi/pulumi/pkg/backend"
+	"github.com/pulumi/pulumi/pkg/backend/display"
+	"github.com/pulumi/pulumi/pkg/diag/colors"
+	"github.com/pulumi/pulumi/pkg/util/cmdutil"
+)
+
+func newHistoryCmd() *cobra.Command {
+	var stack string
+	var outputJSON bool // Requires PULUMI_DEBUG_COMMANDS
+	var cmd = &cobra.Command{
+		Use:        "history",
+		Aliases:    []string{"hist"},
+		SuggestFor: []string{"updates"},
+		Short:      "Update history for a stack",
+		Long: "Update history for a stack\n" +
+			"\n" +
+			"This command lists data about previous updates for a stack.",
+		Args: cmdutil.NoArgs,
+		Run: cmdutil.RunFunc(func(cmd *cobra.Command, args []string) error {
+
+			opts := display.Options{
+				Color: cmdutil.GetGlobalColorization(),
+			}
+
+			s, err := requireStack(stack, false, opts, false /*setCurrent*/)
+
+			if err != nil {
+				return err
+			}
+
+			b := s.Backend()
+
+			updates, err := b.GetHistory(commandContext(), s.Ref())
+			if err != nil {
+				return errors.Wrap(err, "getting history")
+			}
+
+			if outputJSON {
+				b, err := json.MarshalIndent(updates, "", "    ")
+				if err != nil {
+					return err
+				}
+				fmt.Println(string(b))
+			} else {
+				printUpdateHistory(updates, opts)
+			}
+			return nil
+		}),
+	}
+	cmd.PersistentFlags().StringVarP(
+		&stack, "stack", "s", "",
+		"Choose an stack other than the currently selected one")
+	// pulumi/issues/496 tracks adding a --format option across all commands. Rather than expose a partial solution
+	// for just `history`, we put the JSON flag behind an env var so we can use in tests w/o making public.
+	if cmdutil.IsTruthy(os.Getenv("PULUMI_DEBUG_COMMANDS")) {
+		cmd.PersistentFlags().BoolVar(&outputJSON, "output-json", false, "Output stack history as JSON")
+	}
+	return cmd
+}
+func printUpdateHistory(updates []backend.UpdateInfo, opts display.Options) {
+
+	if len(updates) == 0 {
+		fmt.Println("Stack has never been updated")
+		return
+	}
+
+	for _, update := range updates {
+
+		fmt.Print(
+			opts.Color.Colorize(
+				fmt.Sprintf("%s%s%s\n", colors.Green, "-----------------------", colors.Reset)))
```

Comment:
What's  the thinking/intent around this being printed/

Reply:
<img width="1244" alt="updates" src="https://user-images.githubusercontent.com/8211780/49488536-e4aa0600-f7fb-11e8-9ce5-7b44b0d0ef23.png">

Just to make it easier on the user to distinguish between updates. Similar to the activity tab of the service. I can remove this, if you think it's excessive.

Reply:
JSON Output:
<img width="691" alt="json" src="https://user-images.githubusercontent.com/8211780/49491528-62284300-f809-11e8-9a1c-c6ec16deeeee.png">
Normal View :
![stackoutput](https://user-images.githubusercontent.com/8211780/49491532-68b6ba80-f809-11e8-8ae9-5ef4915d0b11.png)

Reply:
Ooh, that looks nice :smile:

Reply:
I think it might make sense to just have a single blank line between entries.  then just colorize 'status: succeeded' and 'status: failed' accordingly.

Reply:
What is 'm:'?  Does it mean 'message'?  If so, i would consider just placing this on another line.

Reply:
<img width="367" alt="nolines" src="https://user-images.githubusercontent.com/8211780/49592355-bde7ee80-f925-11e8-804a-062c1a2bf2ac.png">
Is this better? Any other changes to the formatting that you think the user would like?